src/topp/MapAlignerTreeGuided.cpp

```diff
@@ -0,0 +1,580 @@
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Julia Thueringer $
+// $Authors: $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/APPLICATIONS/MapAlignerBase.h>
+// calculate pearson distance
+#include <OpenMS/MATH/STATISTICS/StatisticFunctions.h>
+// create binary tree
+#include <OpenMS/DATASTRUCTURES/DistanceMatrix.h>
+#include <OpenMS/DATASTRUCTURES/BinaryTreeNode.h>
+#include <OpenMS/COMPARISON/CLUSTERING/ClusterFunctor.h>
+#include <OpenMS/COMPARISON/CLUSTERING/ClusterAnalyzer.h>
+// align maps and generate output
+#include <OpenMS/ANALYSIS/MAPMATCHING/MapAlignmentTransformer.h>
+#include <OpenMS/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmIdentification.h>
+
+using namespace OpenMS;
+using namespace Math;
+using namespace std;
+
+//-------------------------------------------------------------
+// Doxygen docu
+//-------------------------------------------------------------
+
+/**
+  @page TOPP_MapAlignerTree MapAlignerTree
+
+  @brief Corrects retention time distortions between maps by following a tree constructed with SingleLinkage
+
+<CENTER>
+    <table>
+        <tr>
+            <td ALIGN = "center" BGCOLOR="#EBEBEB"> potential predecessor tools </td>
+            <td VALIGN="middle" ROWSPAN=4> \f$ \longrightarrow \f$ MapAlignerTreeGuided \f$ \longrightarrow \f$</td>
+            <td ALIGN = "center" BGCOLOR="#EBEBEB"> potential successor tools </td>
+        </tr>
+        <tr>
+            <td VALIGN="middle" ALIGN = "center" ROWSPAN=2> @ref TOPP_XTandemAdapter @n (or another search engine adapter) </td>
+        </tr>
+        <tr>
+            <td VALIGN="middle" ALIGN = "center" ROWSPAN=2> @ref TOPP_FeatureLinkerUnlabeledKD or @n @ref TOPP_FeatureLinkerUnlabeledQT </td>
+        </tr>
+    </table>
+</CENTER>
+
+    This tool provides an algorithm to align the retention time scales of multiple input files, correcting shifts and distortions between them. Retention time adjustment may be necessary to correct for chromatography differences e.g. before data from multiple LC-MS runs can be combined (feature grouping), or when one run should be annotated with peptide identifications obtained in a different run.
+
+    All map alignment tools (MapAligner...) collect retention time data from the input files and - by fitting a model to this data - compute transformations that map all runs to a common retention time scale. They can apply the transformations right away and return output files with aligned time scales (parameter @p out), and/or return descriptions of the transformations in trafoXML format (parameter @p trafo_out). Transformations stored as trafoXML can be applied to arbitrary files with the @ref TOPP_MapRTTransformer tool.
```

Comment:
`map all runs to a common retention time scale` -> Is that still true here?

Reply:
It's still true: It uses the methods align() and computeTransformation() of the MapAlignmentAlgorithmIdentification class, but determines the order of the alignments of multiple maps based on distances between them and is therefore tree-guided. I will add this to the documentation and also write tests in January.